source/javascripts/view/gallery.js

```diff
@@ -0,0 +1,91 @@
+(function(baseView, loaderView, scrollerView, baseCollection, gridCollection) {
+
+	var NAMESPACE = "wall";
+
+	//
+	// Handle format: landscape/portrait
+	// Set pictures close together
+	// Save picture positions
+	//
+
+	var wallView = function(el, options) {
+		baseView.apply(this, arguments);
+	}
+
+	wallView.prototype = Object.create(baseView.prototype);
+
+	_.extend(wallView.prototype, {
+		
+		collection: new gridCollection(),
+		
+		initialize: function(el, options) {
+			
+			baseView.prototype.initialize.apply(this, arguments);
+			
+			this._fill = this.el.data("fill") || "width";
+			
+			$(this.el).on("load:stop", this.render.bind(this));
+			
+			// @FIXME : put this in the future into render
+			// when all methods will be cleanedup && renamed
+			
+			this.__loader = new loaderView(this.el, {
+				collection: this.collection
+			});
+			this.__loader.render();
+			
+			this.__scroller = new scrollerView(this.el, {
+				collection: this.collection
+			});
+		},
+		
+		render: function(event, options) {
+			
+			var _grid = [];
+			
+			var _all_content = '';
+			
+			var _success = _.debounce(function(data) {
+				this.__scroller.grid = _grid;
+				this.__scroller.render.call(this.__scroller, _all_content);
+				this.trigger("render:finished");
+			}.bind(this), 200);
+			
+			var _render = function(data) {
+				// it depends of screen size
+				// 1 screen height == 2.5 rows
+				var content = '';
+				var model0 = data;
+				var template0 = '<div data-content="<%= type %>">';
```

Comment:
Be careful : test if an element ever exists; if not, create a <div> otherwise use existing element.

See comment : "// Create a new container" into the old file